third_party/move/move-compiler/src/expansion/translate.rs

```diff
@@ -2600,13 +2600,20 @@ fn exp_(context: &mut Context, sp!(loc, pe_): P::Exp) -> E::Exp {
             let discriminator = exp(context, *pd);
             let match_arms = parms
                 .into_iter()
-                .map(|parm| {
+                .filter_map(|parm| {
                     let loc = parm.loc;
                     let (pbl, pc, pb) = parm.value;
-                    let bind_list = bind_list(context, pbl).expect("bind list always present");
-                    let opt_cond = pc.map(|e| *exp(context, e));
-                    let body = *exp(context, pb);
-                    sp(loc, (bind_list, opt_cond, body))
+                    if let Some(bind_list) = bind_list(context, pbl) {
+                        let opt_cond = pc.map(|e| *exp(context, e));
+                        let body = *exp(context, pb);
+                        Some(sp(loc, (bind_list, opt_cond, body)))
+                    } else {
+                        context.env.add_diag(diag!(
+                            Syntax::InvalidLValue,
+                            (loc, "bind list cannot be constructed")
```

Comment:
I wonder if this is always a followup error (like in your test case) whether we should just ignore this if context.env.has_errors, and only produce this error if not.

Reply:
done

Reply:
+1, a brief check of `bind_list` suggest that is the case.  But it would be better if we had a documented policy for these translate functions that they return `None` or `Err` only if a diagnostic has been generated.  Note that
```
    let x: Option<Vec<_>> = Vec::<Option<T>>::new().iter().collect();
```
yields `Some([])`, so that holds for the collector in `bind_list`.  I couldn't find official docs, but https://users.rust-lang.org/t/collect-option-t-on-an-empty-iterator/24966.